tuf/client/updater.py

```diff
@@ -951,18 +951,10 @@ def _import_delegations(self, parent_role):
         # We specify the keyid to ensure that it's the correct keyid
         # for the key.
         try:
+          key, _ = securesystemslib.keys.format_metadata_to_key(keyinfo, keyid)
 
-          # The repo may have used hashing algorithms for the generated keyids
-          # that doesn't match the client's set of hash algorithms.  Make sure
-          # to only used the repo's selected hashing algorithms.
-          hash_algorithms = securesystemslib.settings.HASH_ALGORITHMS
-          securesystemslib.settings.HASH_ALGORITHMS = keyinfo['keyid_hash_algorithms']
-          key, keyids = securesystemslib.keys.format_metadata_to_key(keyinfo)
-          securesystemslib.settings.HASH_ALGORITHMS = hash_algorithms
-
-          for key_id in keyids:
-            key['keyid'] = key_id
-            tuf.keydb.add_key(key, keyid=None, repository_name=self.repository_name)
+          key['keyid'] = keyid
+          tuf.keydb.add_key(key, keyid=None, repository_name=self.repository_name)
```

Comment:
```suggestion
          tuf.keydb.add_key(key, repository_name=self.repository_name)
```
the `keyid` argument defaults to `None`, so let's not pass it here.

Aside: I can't find a single caller of `keydb.add_key()` which passes a value other than `None` for the keyid 😕